examples/interop/RecoveryPlugnWithEndpoints/endpoint.ts

```diff
@@ -0,0 +1,156 @@
+import { BasePluginEndpoint } from 'lisk-sdk';
+import { RecoveryPluginConfig } from './types';
+import { Mainchain } from './Mainchain';
+import { Sidechain } from './Sidechain';
+import { CCMsg } from 'lisk-framework';
+import { getDBInstance } from './db/db';
+import { EventsDb } from './db/EventsDb';
+import { MessageRecoveryDb } from './db/MessageRecoveryDb';
+import { StateRecoveryDB } from './db/StateRecoveryDb';
+
+interface SidechainWithStoreKey {
+	sidechain: Sidechain;
+	storeKeys: string[];
+}
+
+export class Endpoint extends BasePluginEndpoint {
+	private _config!: RecoveryPluginConfig;
+
+	private _messageRecoveryDb!: MessageRecoveryDb;
+	private _stateRecoveryDb!: StateRecoveryDB;
+
+	private _mainchain!: Mainchain;
+	// private _sidechains: Map<string, Sidechain> = new Map();
+	// private _sidechainsWithStoreKeys: Map<string, SidechainWithStoreKey> = new Map();
+	private _sidechainsWithStoreKeys: Map<string, SidechainWithStoreKey> = new Map();
+
+	public load(
+		config: RecoveryPluginConfig,
+		mainchain: Mainchain,
+		messageRecoveryDb: MessageRecoveryDb,
+		stateRecoveryDb: StateRecoveryDB,
+	) {
+		this._config = config;
+		this._mainchain = mainchain;
+		this._messageRecoveryDb = messageRecoveryDb;
+		this._stateRecoveryDb = stateRecoveryDb;
+	}
+
+	/**
+	 * `chainID` MUST correspond to generated sidechain object's `this._nodeInfo.chainID` (otherwise there will be a conflict & can potentially introduce a bug`
+	 * `wsUrl` is needed in case of remote sidechain, since IPCPath only works on local machine
+	 */
+	public async addChainForMessageRecovery(chainID: string, wsUrl: string): Promise<void> {
+		await this._configureSidechain(chainID, wsUrl, null);
+
+		const sidechain = (this._sidechainsWithStoreKeys.get(chainID) as SidechainWithStoreKey)
+			.sidechain;
+		await sidechain.saveProofsForMessageRecovery(this._mainchain.getChainID());
+	}
+
+	public async getPendingCCMs(chainID: Buffer): Promise<CCMsg[]> {
+		const eventsDb = new EventsDb(await getDBInstance());
```

Comment:
why are getting new instance of this DB, why can't we use the one which is already initialized. I think it might not work also if other clients are accessing the same DB